### PR TITLE
add sql state and vendor code for this

### LINKED ISSUE #4654: Throw correct SQLState and vendorCode with SQLException when out of index from ResultSetMetaData.getColumnName()
The current ResultSetMetaData can throw SQLException without SQL State. Please fill the correct SQLState and vendorCode in SQLException.

Please fix on classes under `TODO fill correct SQL state` comment:

- [ ] ShardingResultSetMetaData
- [ ] EncryptResultSetMetaData

### PR DESCRIPTION
**Pull request by zjcnb**
Fixes #4654

### CODE REVIEW

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java [hunk 1]
```diff
@@ -27,4 +27,8 @@
 public final class SQLExceptionConstant {
     
     public static final String SQL_STRING_NULL_OR_EMPTY = "SQL String can not be NULL or empty.";
+    
+    public static final String SQL_STATE = "S1002";
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java]: `SQL_STATE` is a common name, but the value is `S1002`, it is not a good variable name.

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java [hunk 2]
```diff
@@ -27,4 +27,8 @@
...
+    
+    public static final int SQL_VENDOR_CODE = 0;
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java]: Same with `SQL_STATE`

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java (hunk 1 above)
Comment (zjcnb) [reply sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java]: ok

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java [hunk 3]
```diff
@@ -27,4 +27,6 @@
 public final class SQLExceptionConstant {
     
     public static final String SQL_STRING_NULL_OR_EMPTY = "SQL String can not be NULL or empty.";
+    
+    public static final String OUT_OF_INDEX_SQL_STAT = "S1002";
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/constant/SQLExceptionConstant.java]: Please do not use abbreviation as variable name

File: sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java [hunk 4]
```diff
@@ -89,8 +90,7 @@ public String getColumnName(final int column) throws SQLException {
         if (sqlStatementContext instanceof SelectStatementContext) {
             List<Projection> actualProjections = ((SelectStatementContext) sqlStatementContext).getProjectionsContext().getExpandProjections();
             if (column > actualProjections.size()) {
-                // TODO fill correct SQL state
-                throw new SQLException(String.format("Out of index of projection %s", column));
+                throw new SQLException(String.format("Out of index of projection %s", column), SQLExceptionConstant.OUT_OF_INDEX_SQL_STAT, 0);
```
Comment (terrymanu) [inline sharding-jdbc/sharding-jdbc-core/src/main/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/resultset/EncryptResultSetMetaData.java]: Maybe we need to check string information "Out of index of projection %s" too

**Pull request merged**